### PR TITLE
Fixed Multi-selection functionality.

### LINKED ISSUE #9144: Multi selection of investigations not working in Patient log lab results
**Describe the bug**
When Logging the lab results in the Patient investigation tab, we cannot select multiple investigations at once. Atmost we can select 2 (that is by double clicking one option and then clicking on second one, this may also be a bug). 
When selecting multiple investigations from the dropdown in the multiselect form for uploading lab results, the input form for previously selected investigations gets hidden. However, if those forms were filled, their data is still processed during form submission, and empty form also getting submitted.

**To Reproduce**

1. Go to patient tab
2. Select any active patient
3. Go to investigations tab
4. Click on `Log Lab Results`
5. Double click any option from the search drop down and click another option
6. See error
7. Select a option and fill the following form that is given 
8. Select another option from dropdown and see error(previous filled form gets hidden)
9. Submit the form , the data gets processed

**Screenshots**
https://github.com/user-attachments/assets/8e586585-9382-459f-a262-efc2734656bd

**Desktop (please complete the following information):**

- Version 129.0.6668.90 (Official Build) (64-bit)

**Additional context**
I would like to be assigned to this issue and requesting for expected behaviour

### PR DESCRIPTION
**Pull request by Jeffrin2005**
## Proposed Changes

- Fixes #9144
- Enabled the functionality to allow the selection of multiple options.
- Added a Clear button

https://github.com/user-attachments/assets/16ba298c-f15b-4deb-b12e-3bd015f1a65b

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Added a button to clear selected investigations
	- Enhanced investigation selection interface with improved filtering and state management
	- Updated patient investigation testing with advanced filtering scenarios

- **Bug Fixes**
	- Improved error handling for empty investigation parameters
	- Streamlined investigation selection logic

- **UI/UX Improvements**
	- Refined button layout and styling
	- Updated component interaction for better user experience

### CODE REVIEW

File: .gitattributes
Comment (rithviknishad) [inline .gitattributes:1]: revert unrelated changes

File: package-lock.json
Comment (rithviknishad) [inline package-lock.json:1]: revert unrelated changes

**Review by rithviknishad — CHANGES_REQUESTED** [review]
lgtm, but do revert changes made to unrelated files

File: package-lock.json
Comment (Jeffrin2005) [reply package-lock.json:1]: @rithviknishad  I added " * text=auto " to the .gitattributes file. Should I remove it? do I need to update any packages in the package-lock.json file?

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 1]
```diff
@@ -344,23 +342,33 @@ const InvestigationReports = ({ id }: any) => {
...
-                  handleGenerateReports(1);
-                }}
-                disabled={generateReportDisabled}
-                variant="primary"
-                className="my-2.5"
-              >
-                {t("generate_report")}
-              </ButtonV2>
+              <div className="flex space-x-2">
+                <ButtonV2
```
Comment (Jacobjeevan) [inline src/components/Facility/Investigations/Reports/index.tsx]: Switch these out for Shadcn's Button component under ui/

(Switch out all the ButtonV2s in this file while you are at it)

File: package.json [hunk 2]
```diff
@@ -60,10 +60,10 @@
     "@pnotify/mobile": "^5.2.0",
     "@radix-ui/react-dialog": "^1.1.2",
     "@radix-ui/react-dropdown-menu": "^2.1.2",
+    "@radix-ui/react-icons": "^1.3.2",
```
Comment (Jacobjeevan) [inline package.json]: Leave package.json out of the commit.

**Review by Jacobjeevan — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
request for review only once all previously requested changes are addressed

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 3]
```diff
@@ -314,14 +314,13 @@ const InvestigationReports = ({ id }: any) => {
             />
           </div>
           {!isLoading.investigationLoading && (
-            <ButtonV2
+            <button
```
Comment (rithviknishad) [inline src/components/Facility/Investigations/Reports/index.tsx]: not done: https://github.com/ohcnetwork/care_fe/pull/9345#discussion_r1882725431

refer: https://ui.shadcn.com/docs/components/button

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by Jeffrin2005** [comment]
@rithviknishad Used Shadcn Button.

**Comment by Jeffrin2005** [comment]
@rithviknishad Used Shadcn Button. While Using Shadcn Button the "Clear" button turn out to be Black, so modified the buttonPrimary and buttonSecondary. 



https://github.com/user-attachments/assets/08612e38-5111-4480-ac47-e956930321bb

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 4]
```diff
@@ -284,6 +283,11 @@ const InvestigationReports = ({ id }: any) => {
   const prevSessionDisabled = sessionPage <= 1 || isLoading.tableData;
   const nextSessionDisabled = isNextSessionDisabled || isLoading.tableData;
 
+  const buttonPrimary =
+    "bg-green-500 text-white px-4 py-2 rounded hover:bg-green-600 focus:outline-none ";
+  const buttonSecondary =
+    "bg-gray-100 text-gray-500 px-4 py-2 rounded hover:bg-gray-200 focus:outline-none ";
+
```
Comment (rithviknishad) [inline src/components/Facility/Investigations/Reports/index.tsx]: why is this added? just use the variant prop instead
```suggestion
```

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 5]
```diff
@@ -314,14 +318,13 @@ const InvestigationReports = ({ id }: any) => {
             />
           </div>
           {!isLoading.investigationLoading && (
-            <ButtonV2
+            <Button
               onClick={() => fetchInvestigation()}
               disabled={getTestDisabled}
-              variant="primary"
-              className="my-2.5"
+              className={`my-2.5 ${buttonPrimary}`}
```
Comment (rithviknishad) [inline src/components/Facility/Investigations/Reports/index.tsx]:
```suggestion
              variant="primary"
```

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 6]
```diff
@@ -342,23 +345,31 @@ const InvestigationReports = ({ id }: any) => {
...
-                {t("generate_report")}
-              </ButtonV2>
+              <div className="flex space-x-2">
+                <Button
+                  onClick={() => {
+                    setSessionPage(1);
+                    handleGenerateReports(1);
+                  }}
+                  disabled={generateReportDisabled}
+                  className={`my-2.5 ${buttonPrimary}`}
```
Comment (rithviknishad) [inline src/components/Facility/Investigations/Reports/index.tsx]:
```suggestion
                  variant="primary"
```

File: src/components/Facility/Investigations/Reports/index.tsx [hunk 7]
```diff
@@ -342,23 +345,31 @@ const InvestigationReports = ({ id }: any) => {
...
+                >
+                  {t("generate_report")}
+                </Button>
+                <Button
+                  onClick={clearSelectedInvestigations}
+                  disabled={!selectedInvestigations.length}
+                  className={`my-2.5 ${buttonSecondary}`}
```
Comment (rithviknishad) [inline src/components/Facility/Investigations/Reports/index.tsx]:
```suggestion
                  variant="secondary"
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image

@Jeffrin2005 The issue persists in create investigation page

**Comment by Jeffrin2005** [comment]
@nihal467 I couldn't able locate where the issue exists. Could you please tell me where it is?

**Comment by nihal467** [comment]
image

- [ ] click the log update result button and try to multi-select in the dropdown, its part of the investigation module you are trying to fix

**Comment by Jeffrin2005** [comment]
@nihal467 @Jacobjeevan Updated Multiselection in create investigation page

https://github.com/user-attachments/assets/ff3a19f0-2cf9-47af-b773-5de84d10fe9a

**Comment by nihal467** [comment]
@Jeffrin2005 fix the test failure in the investigation test of cypress

**Comment by Jeffrin2005** [comment]
@nihal467 I tried to fix it, but it's been not resolved , Could you help me to resolve it ?

**Comment by Jeffrin2005** [comment]
Fixed cypress test

**Comment by rithviknishad** [comment]
no longer relevant in new architecture

**Pull request closed without merging**